src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/RefVsAnyResult.java

```diff
@@ -0,0 +1,69 @@
+package org.broadinstitute.hellbender.tools.walkers.haplotypecaller;
+
+/**
+ * Holds information about a genotype call of a single sample reference vs. any non-ref event
+ */
+final class RefVsAnyResult {
+    /**
+     * The genotype likelihoods for ref/ref ref/non-ref non-ref/non-ref
+     */
+    private final double[] genotypeLikelihoods;
+
+    /**
+     * AD field value for ref / non-ref
+     */
+    private final int[] AD_Ref_Any = new int[2];
+
+
+    /**
+     * Creates a new ref-vs-alt result indicating the genotype likelihood vector capacity.
+     * @param likelihoodCapacity the required capacity of the likelihood array, should match the possible number of
+     *                           genotypes given the number of alleles (always 2), ploidy (arbitrary) less the genotyping
+     *                           model non-sense genotype count if applies.
+     * @throws IllegalArgumentException if {@code likelihoodCapacity} is negative.
+     */
+    public RefVsAnyResult(final int likelihoodCapacity) {
+        if (likelihoodCapacity < 0) {
+            throw new IllegalArgumentException("likelihood capacity is negative");
+        }
+        genotypeLikelihoods = new double[likelihoodCapacity];
+    }
+
+    /**
+     * @return Get the DP (sum of AD values)
+     */
+    int getDP() { return AD_Ref_Any[0] + AD_Ref_Any[1]; }
+
+    /**
+     * Return the AD fields. Returns the live array.
+     */
+    int[] get_AD_Ref_Any(){
+        return AD_Ref_Any;
+    }
+
+    double[] getGenotypeLikelihoods(){
+        return genotypeLikelihoods;
+    }
+
+    void incrementRefAD(final int by){
+        AD_Ref_Any[0] += by;
+    }
+
+    void incrementNonRefAD(final int by){
+        AD_Ref_Any[1] += by;
+    }
+
+    /**
+     * Cap the het and hom var likelihood values by the hom ref likelihood.
+     */
+    void capByHomRefLikelihood() {
+        final int likelihoodCount = genotypeLikelihoods.length;
+        for (int i = 1; i < likelihoodCount; i++) {
+            genotypeLikelihoods[i] = Math.min(genotypeLikelihoods[0], genotypeLikelihoods[i]);
+        }
+    }
+
+    public void increaseGenotypeLikelihood(final int idx, final double by) {
+        genotypeLikelihoods[idx] += by;
+    }
```

Comment:
The name of the method is hurting my eyes ... does it make sense to talk about "increase" a likelihood? are you make something more likely?... No. In practice is the opposite as you start with 0 and you keep adding negative numbers ...

Looking at the code that use this method; it would be more clear if we change it to simply ```setGenotypeLikelihood```. 
The using code seems to set each likelihood in two steps (thus the need for the "increase" approach"). First increases the numerator and then the denominator. 

Please change the code so that the denominator is calculated at the beginning and then the genotyping setting is only done once.

The code responsible is in ReferenceConfidenceModel#calcGenotypeLikelihoodsOfRefVsAny and ReferenceConfidenceModel#calcPileupElementRefVsNonRefLikelihoodAndCount.

If you want too keep the late denominator calculation (perhaps has a performance advantage) then create a  method to apply such a denominator to all GT likelihoods ```applyGenotypeLikelihoodDenominator```.